JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
+  int nbins{0};
+  for(int i=0; i<hist->GetNbinsX(); i++){
+    double x {hist->GetXaxis()->GetBinCenter(i+1)};
+    if(x>=minX && x<=maxX){
+      xbins[nbins]  = hist->GetXaxis()->GetBinLowEdge(i+1);
+      xbins[nbins+1]= hist->GetXaxis()->GetBinUpEdge(i+1);
+      ++nbins;
+    }
+  }
+
+  TH1F *hnew = new TH1F ("hnew", "", nbins, xbins);
+  for(int i=0; i<nbins+1; i++){
+    hnew->SetBinContent(i+1, hist->GetBinContent(hist->FindBin(hnew->GetBinCenter(i+1))));
+    hnew->SetBinError  (i+1, hist->GetBinError  (hist->FindBin(hnew->GetBinCenter(i+1))));
+  }
+  return hnew;
+}
+
+//______________________________________________________________________________
+void fit_modified_fnc_dscb(TH1F*& hrspOrig, string alg, bool calibrated, string histName){
+
+  TH1F* hrsp=(TH1F*)hrspOrig->Clone(); 
+
+
+  // do no write in the output file all the histogram
+  TH1::AddDirectory(kFALSE);
+
+  // select a particular algorithm
+  //if (alg!="ak5tauHPSlooseCombDBcorrAll") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrOneProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrOneProng1Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrTwoProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrTwoProng1Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrThreeProng0Pi0") return;
+  //if (alg!="ak5tauHPSlooseCombDBcorrThreeProng1Pi0") return;
+  //
+  if (alg=="ak5tauHPSlooseCombDBcorrOneProng2Pi0") return; // =ak5tauHPSlooseCombDBcorrOne
+
+  string histname{hrsp->GetName()};
+
+  // select the relative response histograms for an eta range
+  if ( histname.find(histName) == string::npos) return;
+
+  // select a particular histogram
+  //if ( histname!="RelRsp_JetEta1.7to1.9_RefPt500to3000") return;
+
+  cout << "\nModified CB processing "<<alg<<": " << histname << endl;
+
+  // to avoid that the pol diverge at low value, set empty bin to 0.5+/-0.5
+  if(hrsp->Integral()<300000){
+    for(int i=1; i<=hrsp->GetNbinsX(); i++) if(hrsp->GetBinContent(i)==0){
+      hrsp->SetBinContent(i, 0.5); hrsp->SetBinError(i, 0.5);
+    }
+  }
+
+  ////////////////////////
+  ////////////////////////
+  // start fit procedure
+  ////////////////////////
+  ////////////////////////
+
+  // find the max of the hist that correspond to the tau pic btw [0, 1.05]   
+  double maxBin{0};
+  int binmax = -1;
+  for ( int i = 0; i <= hrsp->FindBin(1.05); ++i ){
+    double newMaxBin = hrsp->GetBinContent(i);
+    if ( newMaxBin > maxBin ) { 
+      maxBin = newMaxBin; binmax = i; 
+    }
+  } 
+
+  // pre-fit function 
+  TF1* testfit{0}; 
+  TF1 *gausLeft{0};
+  TF1 *expLeft{0};
+  TF1 *polLeft{0};
+  // global fit function
+  TF1* modified_fdscb{0};
+
+  // fit definition
+  string gaus = "[0]*TMath::Exp(-0.5*TMath::Power((x-[1])/[2],2.))"; 
+  string expo = "[0]*TMath::Exp([1]*(x - [2]))";
+  string pol = "[0]+[1]*x+[2]*TMath::Power(x,2)+[3]*TMath::Power(x,3)+[4]*TMath::Power(x,4)+[5]*TMath::Power(x,5)+[6]*TMath::Power(x,6)+[7]*TMath::Power(x,7)";
+
```

Comment:
Why use ROOT defined functions here and user defined functions later? It's fine as long as there is a reason, even if it's just that ROOT has limitations.